tst/EnergyPlus/unit/TestHelpers/IOHelper.hh

```diff
@@ -0,0 +1,232 @@
+		int m_MaxNumericIDFArgsFound = 0;
+		int m_MaxAlphaIDFDefArgsFound = 0;
+		int m_MaxNumericIDFDefArgsFound = 0;
+		int m_NumOutOfRangeErrorsFound = 0;
+		int m_NumBlankReqFieldFound = 0;
+		int m_NumMiscErrorsFound = 0;
+		int m_MinimumNumberOfFields = 0;
+		int m_NumObsoleteObjects = 0;
+		int m_TotalAuditErrors = 0;
+		int m_NumSecretObjects = 0;
+		bool m_ProcessingIDD = false;
+		std::string m_InputLine;
+		Array1D_string m_ListOfSections;
+		Array1D_string m_ListOfObjects;
+		Array1D_int m_iListOfObjects;
+		Array1D_int m_ObjectGotCount;
+		Array1D_int m_ObjectStartRecord;
+		std::string m_CurrentFieldName;
+		Array1D_string m_ObsoleteObjectsRepNames;
+		std::string m_ReplacementName;
+		bool m_OverallErrorFlag = false;
+		bool m_EchoInputLine = true;
+		bool m_ReportRangeCheckErrors = true;
+		bool m_FieldSet = false;
+		bool m_RequiredField = false;
+		bool m_RetainCaseFlag = false;
+		bool m_ObsoleteObject = false;
+		bool m_RequiredObject = false;
+		bool m_UniqueObject = false;
+		bool m_ExtensibleObject = false;
+		int m_ExtensibleNumFields = 0;
+		Array1D_bool m_IDFRecordsGotten;
+		Array1D< InputProcessor::ObjectsDefinition > m_ObjectDef;
+		Array1D< InputProcessor::SectionsDefinition > m_SectionDef;
+		Array1D< InputProcessor::FileSectionsDefinition > m_SectionsOnFile;
+		InputProcessor::LineDefinition m_LineItem;
+		Array1D< InputProcessor::LineDefinition > m_IDFRecords;
+		Array1D< InputProcessor::SecretObjects > m_RepObjects;
+		Array1D_string m_cAlphaFieldNames;
+		Array1D_string m_cNumericFieldNames;
+		Array1D_bool m_lNumericFieldBlanks;
+		Array1D_bool m_lAlphaFieldBlanks;
+		Array1D_string m_cAlphaArgs;
+		Array1D< Real64 > m_rNumericArgs;
+	};
+
+	class IOHelper
+	{
+	public:
+		IOHelper();
+		~IOHelper();
+
+		// This sets up the InputProcessor cache specificaly to store the processed full IDD for subsequent runs.
+		static void setup_cache();
+
+		// This first lazy initializes the IDD cache. It is only initialized if it is needed and used by any unit test.
+		// After the first initialization, it uses the cached IDD. The run time of individual unit tests can vary depending on
+		// which unit tests first calls use_cache(), since that first call will take more time.
+		static void use_cache();
+
+		// This will compare either a STL container or ObjexxFCL container
+		// Pass a forward iterator of a container you want to compare against an expected container. You can pass in an existing 
+		// container or use an initializer list like below.
+		// I had to only do one template typename due to wrapping this in Google Test macros causes the macro to fail. It thinks
+		// there are more macro arguments than expected. This is a way around that restriction.
+		// Example Usage: 
+		// 		EXPECT_TRUE( compare_containers< std::vector< bool > >( { true } , ObjectDef( index ).AlphaOrNumeric.begin() ) );
+		// 		EXPECT_TRUE( compare_containers< Array1D_bool >( { true } , ObjectDef( index ).AlphaOrNumeric.begin() ) );
```

Comment:
Yes the compiler could probably deduce if you passed like this. But just using ``{ true }``` it could not.